### PR TITLE
Small fixes to domain whitelist

### PR DESCRIPTION
**Pull request by jerryliu55**
- better default empty whitelist
- unify URL lookups
- display status on add/remove success

### CODE REVIEW

File: src/js/contentScript.js [hunk 1]
```diff
@@ -1,22 +1,18 @@
...
-    whitelist = new Set();
-  } else {
-    whitelist = new Set(result.summaryDomainWhitelist);
-  }
-  var pageDomain = document.domain;
-  if (whitelist.has(pageDomain)) {
+chrome.storage.sync.get({'summaryDomainWhitelist': []}, function(result) {
+  var whitelist = new Set(result.summaryDomainWhitelist);
+
+  var url = new URL(location.href);
```
Comment (kevintpeng) [inline src/js/contentScript.js]: what's location href?

File: src/page_action/popup.js [hunk 2]
```diff
@@ -1,31 +1,39 @@
...
+    var whitelist = new Set(result.summaryDomainWhitelist);
+
+    chrome.tabs.getSelected(null, function(tab) {
       var url = new URL(tab.url)
       whitelist.add(url.hostname);
       chrome.storage.sync.set({summaryDomainWhitelist: [...whitelist]}, function() {
-        app.console.log(whitelist);
+        // display success
+        var statusDisplay = document.getElementById('statusDisplay');
+        statusDisplay.textContent = 'Added "' + url.hostname + '". Refresh to see changes.';
```
Comment (kevintpeng) [inline src/page_action/popup.js]: nice :D

File: src/js/showPageAction.js [hunk 3]
```diff
@@ -4,7 +4,10 @@
 
 // Called when the url of a tab changes.
 function checkForValidUrl(tabId, changeInfo, tab) {
-  chrome.pageAction.show(tabId);
+  var url = new URL(tab.url);
+  if (url.protocol.startsWith("http")) {
```
Comment (kevintpeng) [inline src/js/showPageAction.js]: was this running on the chrome extension pages too or something?

**Comment by kevintpeng** [comment]
should we add a full list of white listed stuff to the options page? should be pretty straight forward I think, but might need to import something to get a nice list with removable elements

File: src/js/contentScript.js (hunk 1 above)
Comment (jerryliu55) [reply src/js/contentScript.js]: it's a google chrome thing that lets you see the current page's url. Before I was doing `document.domain` but that only gave you the base domain

File: src/js/showPageAction.js (hunk 3 above)
Comment (jerryliu55) [reply src/js/showPageAction.js]: yea LOL

**Comment by jerryliu55** [comment]
yea i was thinking that too, made a trello card for it :)

**Pull request merged**